docs/user-guide/src/ipam/introduction.md

```diff
@@ -0,0 +1,138 @@
+
+The IPAM is introduced to manages the allocations of IP subnets according to
+the requests without handling any use of those addresses. The IPAM adds the
+flexibility by providing the address right before provisioning the node. It can
+share a pool across machine deployments or KCP, allow non-continuous pools
+and external IP management by using IPAddress CRs, offer predictable IP
+addresses, and it is resilient to the clusterctl move operation.
+
+In order to use IPAM, both the CAPI controller and IPAM controller are
+required, since the IPAM controller has a dependency on Cluster API *Cluster*
+objects.
+
+## IPAM components
+
+* **IPPool**: A set of IP addresses pools to be used for IP address allocations
+* **IPClaim**: Request for an IP address allocation
+* **IPAddress**: IP address allocation
+
+### IPPool
+
+Example of IPPool:
+
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPPool
+metadata:
+  name: pool1
+  namespace: default
+spec:
+  clusterName: cluster1
+  namePrefix: test1-prov
+  pools:
+    - start: 192.168.0.10
+      end: 192.168.0.30
+      prefix: 25
+      gateway: 192.168.0.1
+    - subnet: 192.168.1.1/26
+    - subnet: 192.168.1.128/25
+  prefix: 24
+  gateway: 192.168.1.1
+  preAllocations:
+    claim1: 192.168.0.12
+```
+
+The *spec* field contains the following fields:
+
+* **clusterName**: Name of the cluster to which this pool belongs, it is used to verify whether the resource is paused.
+* **namePrefix**: The prefix used to generate the IPAddress.
+* **pools**: List of IP address pools
+* **prefix**: Default prefix for this IPPool
+* **gateway**: Default gateway for this IPPool
+* **preAllocations**: Default preallocated IP address for this IPPool
+
+The *prefix* and *gateway* can be overridden per pool. Here is the pool definition:
+
+* **start**: IP range start address and it can be omitted if **subnet** is set.
+* **end**: IP range end address and can be omitted.
+* **subnet**: Subnet for the allocation and can be omitted if **start** is set. It is used to verify that the allocated address belongs to this subnet.
+* **prefix**: Override of the default prefix for this pool
+* **gateway**: Override of the default gateway for this pool
+
+### IPClaim
+
+An IPClaim is an object representing a request for an IP address allocation.
+
+Example of IPClaim:
+
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPClaim
+metadata:
+  name: test1-controlplane-template-0-provisioning-pool
+  namespace: default
+spec:
+  pool:
+    name: pool1
+    namespace: default
+```
+
+The *spec* field contains the following:
+
+* **pool**: This is a reference to the IPPool that is requested for
+
+### IPAddress
+
+An IPAddress is an object representing an IP address allocation. It will be created by IPAM to fill an IPClaim, so that user does not have to create it manually.
+
+Example IPAddress:
+
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPAddress
+metadata:
+  name: pool1-192-168-0-13
```

Comment:
namePrefix defined in the IPPool is: `test1-prov` and IPAddress name should be: `test1-prov-192-168-0-13`